packages/shared/lib/core/governance/utils/isVotingForProposal.ts

```diff
@@ -0,0 +1,25 @@
+import { get } from 'svelte/store'
+
+import { TrackedParticipationOverview } from '@iota/wallet'
+
+import { getParticipationOverview } from '@core/account/api'
+import { selectedAccount } from '@core/account/stores'
+
+import { selectedProposal } from '../stores'
+
+export async function isVotingForProposal(): Promise<boolean> {
+    const overview = await getParticipationOverview(get(selectedAccount)?.index ?? 0)
+    if (overview) {
+        const proposalId = get(selectedProposal)?.id
+        if (proposalId in overview.participations) {
+            const participationOutputs: TrackedParticipationOverview[] = Object.values(
+                overview.participations[proposalId]
+            )
+            return participationOutputs.some((output) => output?.endMilestoneIndex === 0)
+        } else {
+            throw new Error('Unable to retrieve selected proposal in participation overview.')
+        }
+    } else {
+        throw new Error('Unable to retrieve participation overview.')
```

Comment:
Instead of returning errors, you should return false here

Reply:
I think that it is not possible to determine if the selected proposal is being voted on or not in these cases, so returning false is incomplete info, returning false would make it prone to mistakes and harder to debug/understand